jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -281,25 +315,64 @@ boolean isRpcServer() {
       return Tags.SPAN_KIND_SERVER.equals(tags.get(Tags.SPAN_KIND.getKey()));
     }
 
-    @Override
-    public io.opentracing.Span start() {
-      // find preferredParent, it is first childOf
-      Reference preferredParent = references.isEmpty() ? null : references.get(0);
+    private SpanContext preferredParent() {
+      if (parentContext != null) {
+        return parentContext;
+      }
+
+      Reference preferredReference = references.get(0);
       for (Reference reference: references) {
         if (References.CHILD_OF.equals(reference.getType())
-            && !References.CHILD_OF.equals(preferredParent.getType())) {
-          preferredParent = reference;
+            && !References.CHILD_OF.equals(preferredReference.getType())) {
+          preferredReference = reference;
           break;
         }
       }
+      return preferredReference.getSpanContext();
+    }
+
+    private boolean isSampled() {
+      if (parentContext != null && parentContext.isSampled()) {
+        return true;
+      }
+
+      if (references != null) {
+        for (Reference reference : references) {
+          if (reference.getSpanContext().isSampled()) {
+            return true;
+          }
+        }
+      }
+      return false;
+    }
+
+    private String debugId() {
+      if (parentContext != null && parentContext.isDebugIdContainerOnly()) {
+        return parentContext.getDebugId();
+      }
+
+      if (references != null) {
```

Comment:
I think this is over-complicating the use case. The "debug id" (and in the future baggage) is the ability of the tracer to accept some info from the inbound request when the request does not have an actual trace context in it, but the user is supplying debug-id or baggage as a break-glass measure. I guess it's theoretically possible that these break-glass snippets can come to instrumentation that uses FollowsFrom ref type for inbound span, for example a Kafka consumer, but perhaps we can think of that scenario when it actually becomes real. For now child-of is the only case where break-glass data can sneak through.